apps/rpc/src/modules/event/attendance-router.ts

```diff
@@ -136,6 +138,15 @@ export const attendanceRouter = t.router({
       })
     ),
 
+  onRegisterChange: procedure
+    .input(z.object({ attendanceId: AttendanceSchema.shape.id }))
+    .subscription(async function* ({ ctx, signal }) {
+      for await (const [data] of on(ctx.eventEmitter, "attendance:register-change", { signal })) {
+        const attendeeUpdateData = data as { attendee: Attendee; status: "registered" | "deregistered" }
+        yield attendeeUpdateData
+      }
+    }),
+
```

Comment:
This might need some documentation or references, since I have no idea why it is a generator like this or how to make another one

Reply:
Maybe it should also validate the data it gets and throw if its invalid

Reply:
generators are state machines that can yield multiple times

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator